pkg/vault/test/foundry/VaultExplorer.t.sol

```diff
@@ -0,0 +1,750 @@
+        tokenDecimalDiffs[0] = 8;
+        tokenDecimalDiffs[1] = 6;
+
+        _setComplexPoolData();
+
+        feeController = vault.getProtocolFeeController();
+        feeControllerAuth = IAuthentication(address(feeController));
+
+        waDAI = new ERC4626TestToken(dai, "Wrapped aDAI", "waDAI", 18);
+
+        explorer = new VaultExplorer(vault);
+    }
+
+    function testGetVaultContracts() public view {
+        assertEq(explorer.getVault(), address(vault), "Vault address mismatch");
+        assertEq(explorer.getVaultExtension(), vault.getVaultExtension(), "Vault Extension address mismatch");
+        assertEq(explorer.getVaultAdmin(), vault.getVaultAdmin(), "Vault Admin address mismatch");
+        assertEq(explorer.getAuthorizer(), address(vault.getAuthorizer()), "Authorizer address mismatch");
+        assertEq(
+            explorer.getProtocolFeeController(),
+            address(vault.getProtocolFeeController()),
+            "Protocol Fee Controller address mismatch"
+        );
+    }
+
+    function testPoolTokenCount() public view {
+        (uint256 tokenCountVault, uint256 tokenIndexVault) = vault.getPoolTokenCountAndIndexOfToken(pool, dai);
+        (uint256 tokenCountExplorer, uint256 tokenIndexExplorer) = explorer.getPoolTokenCountAndIndexOfToken(pool, dai);
+
+        assertEq(tokenCountExplorer, tokenCountVault, "Token count mismatch");
+        assertEq(tokenIndexExplorer, tokenIndexVault, "Token index mismatch");
+    }
+
+    function testUnlocked() public {
+        assertFalse(explorer.isUnlocked(), "Should be locked");
+
+        vault.manualSetIsUnlocked(true);
+        assertTrue(explorer.isUnlocked(), "Should be unlocked");
+    }
+
+    function testNonzeroDeltaCount() public {
+        assertEq(explorer.getNonzeroDeltaCount(), 0, "Wrong initial non-zero delta count");
+
+        vault.manualSetNonZeroDeltaCount(47);
+        assertEq(explorer.getNonzeroDeltaCount(), 47, "Wrong non-zero delta count");
+    }
+
+    function testGetTokenDelta() public {
+        assertEq(vault.getTokenDelta(dai), 0, "Initial token delta non-zero (Vault)");
+        assertEq(explorer.getTokenDelta(dai), 0, "Initial token delta non-zero (Explorer)");
+
+        dai.mint(address(vault), defaultAmount);
+
+        vault.manualSetIsUnlocked(true);
+        uint256 settlementAmount = vault.settle(dai, defaultAmount);
+        int256 vaultDelta = vault.getTokenDelta(dai);
+
+        assertEq(settlementAmount, defaultAmount, "Wrong settlement amount");
+        assertEq(vaultDelta, -settlementAmount.toInt256(), "Wrong Vault token delta");
+        assertEq(explorer.getTokenDelta(dai), vaultDelta, "getTokenDelta mismatch");
+    }
+
+    function testGetReservesOf() public {
+        dai.mint(address(vault), defaultAmount);
+
+        assertEq(vault.getReservesOf(dai), defaultAmount, "Wrong Vault reserves");
+        assertEq(explorer.getReservesOf(dai), defaultAmount, "Wrong Explorer reserves");
```

Comment:
reserves are actually 0 here :smile: 

you'd need to unlock and settle for the vault to acknowledge them